File: algorithms/66.py
# https://leetcode.com/problems/plus-one


class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """

        i, adding = len(digits) - 1, 1
        while i >= 0 and adding != 0:
            d = digits[i] + adding
            (digits[i], adding) = (d, 0) if d < 10 else (0, 1)
            i -= 1

        if adding != 0:
            digits.insert(0, 1)
        return digits

File: algorithms/66.swift
// https://leetcode.com/problems/plus-one

class Solution {
    func plusOne(digits: [Int]) -> [Int] {
        var ds = digits
        for i in (ds.count-1).stride(to: -1, by: -1) {
            if ds[i] < 9 {
                ds[i] += 1
                return ds
            }
            ds[i] = 0
        }
        ds.insert(1, atIndex: 0)
        return ds
    }
}
